platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java

```diff
@@ -97,11 +109,42 @@ public Response callback(@Context UriInfo uriInfo) throws OAuthAuthenticationExc
         final String providerName = getParameter(params, "oauth_provider");
         OAuthAuthenticator oauth = getAuthenticator(providerName);
         final List<String> scopes = params.get("scope");
-        final String userId = oauth.callback(requestUrl, scopes == null ? Collections.<String>emptyList() : scopes);
+        oauth.callback(requestUrl, scopes == null ? Collections.<String>emptyList() : scopes);
         final String redirectAfterLogin = getParameter(params, "redirect_after_login");
         return Response.temporaryRedirect(URI.create(redirectAfterLogin)).build();
     }
 
+    /**
+     * Gets list of installed OAuth authenticators.
+     *
+     * @param uriInfo UriInfo
+     * @return  list of installed OAuth authenticators
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Set<OAuthAuthenticatorDescriptor> getRegisteredAuthenticators(@Context UriInfo uriInfo) {
```

Comment:
You can move UriInfo to private class members to avoid ugly parameter of this method